features/telemetry/daemon/src/main/java/org/opennms/netmgt/telemetry/daemon/TelemetryMessageConsumer.java

```diff
@@ -99,34 +106,59 @@ public void handleMessage(TelemetryProtos.TelemetryMessageLog messageLog) {
     }
 
     private Adapter buildAdapter(org.opennms.netmgt.telemetry.config.model.Adapter adapterDef) throws Exception {
```

Comment:
I think we should move all the logic about constructing an adapter into the registry instead of having it here.

Reply:
Took care of all the review comments.